File: Python/Arrays/L_buildArrWStackOp.py
from typing import List


def buildArray(target: List[int], n: int) -> List[str]:
    stack = []
    m = len(target)
    currIdx = 0
    for i in range(1,n+1):
        if i == target[currIdx]:
            stack.append("Push")
            currIdx += 1
        else:
            stack.append("Push")
            stack.append("Pop") 
        if currIdx == m:
            break
    return stack


print(buildArray([1,3], 3))

File: cpp/Array/L_buildArrWStackOp.cpp
#include <bits/stdc++.h>
using namespace std;

vector<string> buildArray(vector<int> &target, int n)
{
    unordered_set<int> target_set(target.begin(), target.end());
    vector<std::string> result;

    for (int i = 1; i <= target.back(); ++i)
    {
        if (target_set.find(i) != target_set.end())
        {
            result.push_back("Push");
        }
        else
        {
            result.push_back("Push");
            result.push_back("Pop");
        }
    }
    return result;
}